crates/utils/re_video/src/demux/mod.rs

```diff
@@ -1089,4 +1082,97 @@ mod tests {
         // (this is not the last element in the list since that one doesn't have the highest PTS)
         assert_eq!(Some(48), query_pts(Time(123123123123123123)));
     }
+
+    /// Helper function to check if data contains Annex B start codes
+    fn has_annexb_start_codes(data: &[u8]) -> bool {
+        const ANNEXB_NAL_START_CODE: &[u8] = &[0x00, 0x00, 0x00, 0x01];
+        data.windows(4).any(|w| w == ANNEXB_NAL_START_CODE)
+    }
+
+    fn video_test_file_mp4(codec: VideoCodec, need_dts_equal_pts: bool) -> std::path::PathBuf {
+        let workspace_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .parent()
+            .and_then(|p| p.parent())
+            .and_then(|p| p.parent())
+            .unwrap()
+            .to_path_buf();
+
+        let codec_str = match codec {
+            VideoCodec::H264 => "h264",
+            VideoCodec::H265 => "h265",
+            VideoCodec::VP9 => "vp9",
+            VideoCodec::VP8 => {
+                panic!("We don't have test data for vp8, because Mp4 doesn't support vp8.")
+            }
+            VideoCodec::AV1 => "av1",
+        };
+
+        if need_dts_equal_pts && (codec == VideoCodec::H264 || codec == VideoCodec::H265) {
+            // Only H264 and H265 have DTS != PTS when b-frames are present.
+            workspace_dir.join(format!(
+                "tests/assets/video/Big_Buck_Bunny_1080_1s_{codec_str}_nobframes.mp4",
+            ))
+        } else {
+            workspace_dir.join(format!(
+                "tests/assets/video/Big_Buck_Bunny_1080_1s_{codec_str}.mp4",
+            ))
+        }
+    }
+
+    /// Helper function to test video sampling for a specific codec
+    fn test_video_codec_sampling(codec: VideoCodec, need_dts_equal_pts: bool) {
+        let video_path = video_test_file_mp4(codec, need_dts_equal_pts);
+        let data = std::fs::read(&video_path).unwrap();
+        let video_data = VideoDataDescription::load_from_bytes(
+            &data,
+            "video/mp4",
+            &format!("test_{codec:?}_video_sampling"),
+        )
+        .unwrap();
+
+        let buffers: crate::StableIndexDeque<&[u8]> = std::iter::once(data.as_slice()).collect();
+
+        let mut idr_count = 0;
+        let mut non_idr_count = 0;
+
+        const ANNEXB_NAL_START_CODE: &[u8] = &[0x00, 0x00, 0x00, 0x01];
```

Comment:
how often do we have this constant by now? two times alone in this the diff of this file 😛